libraries/SITL/SIM_Aircraft.cpp

```diff
@@ -898,7 +898,7 @@ void Aircraft::smooth_sensors(void)
  */
 float Aircraft::filtered_idx(float v, uint8_t idx)
 {
-    if (sitl->servo_speed <= 0) {
+    if (sitl == nullptr || sitl->servo_speed <= 0) {
```

Comment:
my guess is these changes are so that examples build in SITL. Should we add examples to run under CI, to ensure examples stay working under sitl.

Reply:
we check they build in CI now, what we don't do is run them. Running them is tricky as they just produce terminal output and run forever, so pass condition isn't obvious

Reply:
i've done it in another way that is less intrusive